client/src/main/java/com/tigrisdata/db/client/TigrisDatabase.java

```diff
@@ -91,6 +92,12 @@ <T extends TigrisCollectionType> DropCollectionResponse dropCollection(Class<T>
    */
   DatabaseDescription describe() throws TigrisException;
 
+  /**
+   * @return stream of JSON events.
+   * @throws TigrisException in case of an error.
+   */
+  Iterator<StreamEvent> stream() throws TigrisException;
```

Comment:
Looks great overall.
can you please add test for this and the other async implementation?

Reply:
I'll be adding tests later, just making it work for now per priority order discussion